### PR TITLE
Preset: enforce Airbnb rule 7.11 (function signatures)

### PR DESCRIPTION
**Pull request by redoPop**
This PR addresses some disagreements between [`airbnb.json`](https://github.com/jscs-dev/node-jscs/blob/c704e9e0713c5462a61dac309a6e6abd30098545/presets/airbnb.json) and [Airbnb rule 7.11](https://github.com/airbnb/javascript/blob/c2fd8fbbd6cf058244024db3ab791ddba15567a1/README.md#7.11):

```javascript
// bad
const f = function(){};
const g = function (){};
const h = function() {};

// good
const x = function () {};
const y = function a() {};
```

* Spaces before opening round braces in anonymous function expressions should be required, not [disallowed](https://github.com/jscs-dev/node-jscs/blob/c704e9e0713c5462a61dac309a6e6abd30098545/presets/airbnb.json#L9-L11)
	* Shorthand is an exception, per [Airbnb rule 3.5](https://github.com/airbnb/javascript/blob/c2fd8fbbd6cf058244024db3ab791ddba15567a1/README.md#3.5)
* Due to the need for differentiated behavior between named vs. anonymous function expressions, the broader rule `disallowSpacesInFunctionExpression` should _not_ be set

I've updated the preset test file (`airbnb.js`) accordingly.

### CODE REVIEW

File: presets/airbnb.json [hunk 1]
```diff
@@ -3,12 +3,6 @@
     "disallowSpacesInNamedFunctionExpression": {
```
Comment (hzoo) [inline presets/airbnb.json]: Sounds like we should remove this as well right?

Comment (redoPop) [inline presets/airbnb.json]: I think this rule should be retained; named function expressions seem to be handled differently from anonymous ones in Airbnb's style guide:

```javascript
// good
const x = function () {};
const y = function a() {};
```

Where `x` is considered an example of an anonymous function expression (where the space before open paren is required) and `y` is considered a named function expression (where the space before open paren is disallowed).

**Comment by markelog** [comment]
@hzoo LGTM?

**Comment by hzoo** [comment]
Yeah :+1: - Thanks @redoPop!

**Pull request closed without merging**